collect_app/src/test/java/org/odk/collect/android/activities/FirstLaunchActivityTest.kt

```diff
@@ -1,57 +1,54 @@
-package org.odk.collect.android.fragments.dialogs
+package org.odk.collect.android.activities
 
 import android.content.Context
+import androidx.test.core.app.ActivityScenario
 import androidx.test.core.app.ApplicationProvider
-import androidx.test.espresso.Espresso.onView
 import androidx.test.espresso.action.ViewActions.click
-import androidx.test.espresso.action.ViewActions.pressBack
 import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.Espresso.onView
 import androidx.test.espresso.intent.Intents
 import androidx.test.espresso.intent.Intents.intended
 import androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent
 import androidx.test.espresso.matcher.ViewMatchers.assertThat
 import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
-import androidx.test.espresso.matcher.ViewMatchers.isRoot
 import androidx.test.espresso.matcher.ViewMatchers.withId
 import androidx.test.espresso.matcher.ViewMatchers.withText
 import androidx.test.ext.junit.runners.AndroidJUnit4
 import org.hamcrest.CoreMatchers.`is`
 import org.hamcrest.CoreMatchers.notNullValue
-import org.junit.Test
 import org.junit.runner.RunWith
+import org.junit.Test
 import org.mockito.Mockito.`when`
 import org.mockito.Mockito.mock
 import org.mockito.Mockito.verify
-import org.odk.collect.android.R
-import org.odk.collect.android.activities.MainMenuActivity
 import org.odk.collect.android.application.Collect
 import org.odk.collect.android.injection.config.AppDependencyModule
 import org.odk.collect.android.preferences.source.SettingsProvider
 import org.odk.collect.android.projects.AutomaticProjectCreatorDialog
 import org.odk.collect.android.projects.ManualProjectCreatorDialog
 import org.odk.collect.android.projects.ProjectImporter
+import org.odk.collect.android.R
 import org.odk.collect.android.storage.StoragePathProvider
 import org.odk.collect.android.support.CollectHelpers
 import org.odk.collect.android.utilities.TranslationHandler.getString
 import org.odk.collect.android.version.VersionInformation
 import org.odk.collect.projects.ProjectsRepository
-import org.odk.collect.testshared.RobolectricHelpers
 
 @RunWith(AndroidJUnit4::class)
-class FirstLaunchDialogTest {
-
+class FirstLaunchActivityTest {
     @Test
-    fun `The dialog should not be dismissed after clicking on a device back button`() {
-        val scenario = RobolectricHelpers.launchDialogFragment(FirstLaunchDialog::class.java, R.style.Theme_Collect_Light)
-        scenario.onFragment {
-            assertThat(it.isVisible, `is`(true))
-            onView(isRoot()).perform(pressBack())
-            assertThat(it.isVisible, `is`(true))
+    fun `The main menu should be started after clicking on the 'Try a demo' link`() {
+        val scenario = ActivityScenario.launch(FirstLaunchActivity::class.java)
```

Comment:
I'd prefer to use a rule rather than explicitly building the scenario every time but I can't figure out how to do the dependency override in that case. Ok to leave like this for now?